src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react"
+import ArrowBackIosNewRoundedIcon from "@mui/icons-material/ArrowBackIosNewRounded"
+import CancelRoundedIcon from "@mui/icons-material/CancelRounded"
+import { IconButton, Typography } from "@mui/material"
+import { ModalProps } from "./Modal.types"
+import {
+  ModalActionsContainer,
+  ModalContainer,
+  ModalContentContainer,
+  ModalTitleContainer,
+} from "./Modal.styled"
+
+const ModalContent = React.memo(
+  (
+    props: Partial<
+      Pick<
+        ModalProps,
+        | "header"
+        | "children"
+        | "actions"
+        | "title"
+        | "subtitle"
+        | "size"
+        | "onClose"
+        | "onBack"
+      >
+    >
+  ) => {
+    const {
+      children,
+      header,
+      actions,
+      title,
+      subtitle,
+      size,
+      onClose,
+      onBack,
+    } = props
+
+    return (
+      <ModalContainer elevation={24} size={size}>
```

Comment:
Has MUi a custom component for modals? Why aren't we using that instead of our own?